### PR TITLE
529 event trigger control

### LINKED ISSUE #529: Encounter status event scheduling
Add control to schedule a status event into the encounter context
- [ ] Must have on trigger to set it active (e.g. based on field set or unset)
- [ ] Show "effective" encounter status in the encounter list view

Lab report example:
- When lab testing has been done set the status to "Lab report pending".
- When lab report is entered set status to "Lab report received".
To achieve this add two controls, one to trigger the the pending status and the second to trigger the received status.

### PR DESCRIPTION
**Pull request by clemens-msupply**
Closes #529 

- Rename enrolment status to label and similar an encounter has a label and a status now (suggestions?)

Testing:
- In the testing program fill the HIV Testing section -> the encounter label should now be "Pending Lab Result"
- Fill in the HIV Result section of this encounter -> label should now be "Lab Result"

### CODE REVIEW

File: client/packages/common/src/intl/locales/en/common.json [hunk 1]
```diff
@@ -141,6 +139,7 @@
   "label.items-selected_other": "{{count}} items selected",
   "label.line": "Line {{line, number}}",
   "label.line-total": "Line total",
+  "label.label": "Label",
```
Comment (mark-prins) [inline client/packages/common/src/intl/locales/en/common.json:142]: tbh - I don't find `Label` helpful - it doesn't tell me what the intention of the column is! Is `Status` not correct here?

**Review by mark-prins — APPROVED** [review]
👍  working nicely!

( note: I've added a commit to fix a postgres script error )

File: client/packages/system/src/Encounter/DetailView/Footer.tsx [hunk 2]
```diff
@@ -31,7 +43,20 @@ export const Footer: FC<FooterProps> = ({ isDisabled, onCancel, onSave }) => {
           alignItems="center"
           height={64}
         >
-          <Box flex={1} display="flex" justifyContent="flex-end" gap={2}>
+          <ButtonWithIcon
```
[the remark this answers is not in the document]
Comment (clemens-msupply) [reply client/packages/system/src/Encounter/DetailView/Footer.tsx:46]: Yes currently just a dev feature...

File: client/packages/common/src/intl/locales/en/common.json (hunk 1 above)
Comment (clemens-msupply) [reply client/packages/common/src/intl/locales/en/common.json:142]: The encounter list already has a status column, I could somehow merge them but in general there could be multiple statuses valid at the same time...

File: client/packages/system/src/Patient/JsonForms/components/EventTrigger.tsx [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+  context?: string;
+}
+
+const scheduleEvent = (
+  event: OptionEvent,
+  baseTime: Date,
+  group?: string,
+  context?: string
+): EncounterEvent => {
+  const datetimePlusMin = DateUtils.addMinutes(
```
[the remark this answers is not in the document]
Comment (clemens-msupply) [reply client/packages/system/src/Patient/JsonForms/components/EventTrigger.tsx]: baseDate is actually baseDatetime but I improved it as suggested

**Pull request merged**